app/controllers/harvests_controller.rb

```diff
@@ -12,16 +12,29 @@ def create
     authorize! :create, DlmeJson
 
     upload = NdjsonUpload.new(params.require(:url))
-
-    return invalid_file unless upload.valid?
+    return handle_error(upload.error) unless upload.valid?
 
     AddResourcesJob.perform_later upload.filepath, exhibit: current_exhibit, local: true
 
     redirect_to spotlight.admin_exhibit_catalog_path(current_exhibit),
                 notice: t('spotlight.resources.fetch.queued')
   end
 
-  def invalid_file
-    redirect_to spotlight.new_exhibit_resource_path(current_exhibit), flash: { error: t('dlme_s3s.form.error') }
+  def handle_error(error)
+    error_message = error_message(error)
+    redirect_to spotlight.new_exhibit_resource_path(current_exhibit), flash: { error: error_message }
+  end
+
+  def error_message(error)
+    # NdjsonUpload can also return a ":no_url" error, but the harvest controller requires the url,
+    # so we do not need to check for that situation
+    case error
+    when :duplicate_ids
+      t('dlme_s3s.form.error_duplicate_ids')
```

Comment:
Would you mind moving these to `".error.#{error}"`?  It doesn't make sense to talk about `S3` anymore since we're trying to decouple from AWS.

Reply:
I did have a comment about that in the issue and thought maybe that could be a different pull request, but I can make that change here.

Reply:
There is a whole dlme_s3s section which may need to go away in general, but maybe I can hold off on that and just move these errors first

Reply:
Thanks for walking me through it! Changes in!